tests/shouldwork/Numbers/Exp.hs

```diff
@@ -3,6 +3,7 @@
 {-# LANGUAGE PartialTypeSignatures #-}
 
 {-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+{-# OPTIONS_GHC -fno-specialise #-} -- TODO: investigate GHC bug
```

Comment:
Report this as an issue, this seems to weird to let slip by

Reply:
I think it might be related to https://gitlab.haskell.org/ghc/ghc/-/issues/23109#note_499130